cpp/include/cuspatial/detail/utility/device_atomics.cuh

```diff
@@ -260,5 +264,24 @@ __device__ inline float atomicMax(thrust::device_ptr<float> ptr, float val)
   return atomicMax(thrust::raw_pointer_cast(ptr), val);
 }
 
+/**
+ * @brief Factory method to create atomic_ref from a thrust::device_reference
+ */
+template <typename T>
+auto __device__ make_atomic_ref(thrust::device_reference<T> ref)
+{
+  T& raw_ref = thrust::raw_reference_cast(ref);
+  return cuda::atomic_ref<T>{raw_ref};
+}
+
+/**
+ * @brief Factory method to create atomic_ref from raw reference
+ */
+template <typename T>
+auto __device__ make_atomic_ref(T& ref)
+{
+  return cuda::atomic_ref<T>{ref};
```

Comment:
This will default to the widest thread scope of `thread_scope_system`, which is likely a wider scope than you want.  You'd only use that if you were concurrently doing atomic operations from both host/device threads, or device threads across different devices. 

I'm guessing `thread_scope_device` is what you want here. 

More info: https://nvidia.github.io/libcudacxx/extended_api/thread_scopes.html


```suggestion
    return cuda::atomic_ref<T, cuda::thread_scope_device>{ref};
```

Reply:
Perhaps make this a template parameter of the factory.